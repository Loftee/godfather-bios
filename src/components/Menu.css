.header {
    background-color: rgba(0, 0, 0, 0.3);
    min-height: 100vh;
    float: left;
    width: 25%;
    padding-bottom: 30px;
}

.header__logo {
    width: 100%;
    height: 300px;
    display: table;
}

.header__logo img {
    width: 170px;
    display: table-cell;
    vertical-align: middle;
    height: 100%;
    margin: 0 auto;
}

.header__menu {
    list-style: none;
    padding: 0;
    margin: 60px 0 0;
}

.header__menu > li {
    width: 100%;
}

.header__menu-link {
    color: white;
    display: block;
    width: 100%;
    text-align: center;
    padding: 10px 0;
    text-decoration: none;
}

.header__menu-link:hover,
.header__menu-link--active {
    color: #40d2fa;
}

.header__menu-link:hover {
    background: #25252d;
}

@media (min-width:768px) {
    .header {
        max-width: 380px;
        align-items: stretch;
    }
}

@media (min-width: 768px) and (max-width: 1199px) {
    .header {
        width: 40%;
    }
}

@media screen and (max-width: 767px) {
    .header {
        width: 100%;
        min-height: initial;
    }
}