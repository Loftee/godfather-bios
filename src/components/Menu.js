import React, {Component} from 'react';
import Logo from '../assets/images/the-godfather.svg';

import './Menu.css';

class Menu extends Component {
    render() {
        const {employees, selectEmployee} = this.props;
        return (
            <div className={'header'}>
                <div className={'header__logo'}>
                    <img src={Logo} alt="The Godfather"/>
                </div>
                <ul className={'header__menu'}>
                    {
                        employees.map((employee, index) => {
                                return (
                                    <li key={index}>
                                        <a
                                            style={{
                                                fontSize: (employee.popularity * 0.5) * 14
                                            }}
                                            className={employee.selected ? 'header__menu-link header__menu-link--active' : 'header__menu-link'}
                                            href={'#'+ employee.name}
                                            onClick={event => selectEmployee(event, employee.name)}
                                        >{employee.name}</a>
                                    </li>
                                )
                            }
                        )
                    }
                </ul>
            </div>
        )
    }
}

export default Menu;