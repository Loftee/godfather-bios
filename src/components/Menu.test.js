import React from 'react';
import {shallow} from "enzyme";
import Menu from './Menu';

describe('Menu Component', () => {
   it('renders correctly', () => {
      const component = shallow(<Menu />);
      expect(component).toMatchSnapshot();
   });
});