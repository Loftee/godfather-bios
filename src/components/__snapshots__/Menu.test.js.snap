// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="header__logo"
  >
    <img
      alt="The Godfather"
      src="the-godfather.svg"
    />
  </div>
</div>
`;
